### ISSUE TITLE
SQLite double.MaxValue rounding issue?

### ISSUE THREAD

**Issue by HypeillingerChillinger** [linq2db/linq2db#2354]
"Where" does not find an entry that does exist. Is there a rounding issue? May be there is somewhere a doubleValue.ToString("G15", CultureInfo.InvariantCulture) missing?

```c#
public class MyClass
{
    public double MyProperty {get;set;}
}


var context = new DataConnection(new SQLiteDataProvider(ProviderName.SQLiteClassic), "Data Source=:memory:");
context.CreateTable<MyClass>();


var instance = new MyClass();
instance.MyProperty = double.MaxValue;

context.Insert(instance);

var queryable = context.GetTable<MyClass>();

// count == 1  => contains the entry with the correct value 1.7976931348623157E+308
// found here https://docs.microsoft.com/de-de/dotnet/api/system.double.maxvalue?view=netcore-3.1
var all = queryable.ToList(); 

// empty result 
var result = queryable.Where(x => x.MyProperty == double.MaxValue).ToList();
```

**Comment by MaceWindu** [comment]
What sql generated for your insert and select queries? If it use parameters, it should work (we have roundtrip tests) for it. For literal it will not work, because sqlite double parser is not precise. See issue I raised couple of years ago https://system.data.sqlite.org/index.html/tktview?name=fb9e4b3087

**Comment by HypeillingerChillinger** [comment]
The query ends with

WHERE [x].[MyProperty] = 1.7976931348623157E+308

Which is correct. So the problem is SQLite!?

**Comment by MaceWindu** [comment]
I see, if you didn't enables parameters inlining, it should generate parameters. Probably will be fixed with #2347 next release.
@sdanyliv  could you confirm it will cover this case too?

**Comment by sdanyliv** [comment]
`double.MaxValue` is constant, so we generated constant and it is usual behavior in LINQ.
Try to assign to variable first:
```cs
var value = double.MaxValue;
var result = queryable.Where(x => x.MyProperty == value).ToList();
```

**Comment by HypeillingerChillinger** [comment]
That works for the example above. 
Actually I add the given value via Expression.Constant. Is it possible to achive the usage of parameters with Expression.Constant?

**Comment by sdanyliv** [comment]
Not yet, but do the following: create dummy class which holds value and pass not Expression.Constant but MemberExpression.
```
var holder = new Holder<double>(double.MaxValue);
var objExpr = Expression.Constant(holder);
var valueExpr = Expression.PropertyOrField(objExpr, "Value");
```

**Comment by sdanyliv** [comment]
I forgot about already present class for that - Tuple:
```cs
var holder    = new Tuple<double>(double.MaxValue);
var objExpr   = Expression.Constant(holder);
var valueExpr = Expression.PropertyOrField(objExpr, "Item1");
```

**Comment by HypeillingerChillinger** [comment]
Ok, that works. I will track  #2347 and remove that workaround.

Now I observed another issue. Please tell me if I should add another github issue.

```c#
public class MyClass
{
    public ulong? MyProperty {get;set;}
}


var context = new DataConnection(new SQLiteDataProvider(ProviderName.SQLiteClassic), "Data Source=:memory:");
context.CreateTable<MyClass>();


var instance = new MyClass();
instance.MyProperty = ulong.MaxValue;

context.Insert(instance);

var queryable = context.GetTable<MyClass>();

// fails => System.OverflowException
var all = queryable.ToList(); 


```

**Comment by jogibear9988** [comment]
should be fixed by #2347
